src/Service/RegistrationService.php

```diff
@@ -0,0 +1,96 @@
+<?php
+
+/**
+ * Copyright 2014 SURFnet bv
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Surfnet\StepupU2fBundle\Service;
+
+use Surfnet\StepupU2fBundle\Dto\RegisterRequest;
+use Surfnet\StepupU2fBundle\Dto\RegisterResponse;
+use Surfnet\StepupU2fBundle\Dto\Registration;
+use Surfnet\StepupU2fBundle\Exception\LogicException;
+use u2flib_server\Error;
+use u2flib_server\RegisterRequest as YubicoRegisterRequest;
+use u2flib_server\U2F;
+
+final class RegistrationService
+{
+    /**
+     * @var \u2flib_server\U2F
+     */
+    private $u2fService;
+
+    public function __construct(U2F $u2fService)
+    {
+        $this->u2fService = $u2fService;
+    }
+
+    /**
+     * @return RegisterRequest
+     */
+    public function requestRegistration()
+    {
+        /** @var YubicoRegisterRequest $yubicoRequest */
+        list($yubicoRequest) = $this->u2fService->getRegisterData();
```

Comment:
Other than the not-enough-entropy exception, which I'm letting bubble up, and some `InvalidArgumentException`s which I cannot trigger, no.